src/auth/bearer/fake/fake.cc

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2014, Treehouse Networks Ltd. New Zealand
```

Comment:
AFAIK, authors for the new code should be listed in CONTRIBUTORS. Exceptions to that Project policy are certainly possible, but, being exceptions, they naturally require a justification. Since I saw no reason for an exception here, and you did not volunteer a reason, I asked for it. IMO, you have not answered that question yet.

N.B. Please note that my question is about the custom copyright statement, not the license. You have already explained why a second license is desirable in this case, and I do not object to adding that license.